app/workers/mail_chimp_worker.rb

```diff
@@ -0,0 +1,21 @@
+class MailChimpWorker
+  include Sidekiq::Worker
+
+  def perform(user_id)
+    user= User.find(user_id)
+    subscribe_to_mailchimp(ENV['USERS_LIST_ID'], user)
+    if user_id.subscribe_newsletter
+      subscribe_to_mailchimp(ENV['NEWSLETTER_LIST_ID'], user)
+    end
+  end
+
+  def subscribe_to_mailchimp(list_id, user)
```

Comment:
@amrtanair make sure you test this method locally using your own mailchimp account and going through the signup process.